sdk/python/feast/stream_feature_view.py

```diff
@@ -1,32 +1,112 @@
+import abc
+import functools
+import warnings
 from datetime import timedelta
+from types import MethodType
 from typing import Dict, List, Optional, Union
 
-from feast.data_source import DataSource
+import dill
+from google.protobuf.duration_pb2 import Duration
+
+from feast.data_source import DataSource, KafkaSource
 from feast.entity import Entity
 from feast.feature_view import FeatureView
 from feast.field import Field
 from feast.protos.feast.core.DataSource_pb2 import DataSource as DataSourceProto
+from feast.protos.feast.core.OnDemandFeatureView_pb2 import (
+    UserDefinedFunction as UserDefinedFunctionProto,
+)
+from feast.protos.feast.core.StreamFeatureView_pb2 import (
+    Aggregation as AggregationProto,
+)
+from feast.protos.feast.core.StreamFeatureView_pb2 import (
+    StreamFeatureView as StreamFeatureViewProto,
+)
+from feast.protos.feast.core.StreamFeatureView_pb2 import (
+    StreamFeatureViewMeta as StreamFeatureViewMetaProto,
+)
+from feast.protos.feast.core.StreamFeatureView_pb2 import (
+    StreamFeatureViewSpec as StreamFeatureViewSpecProto,
+)
+
+warnings.simplefilter("once", RuntimeWarning)
+
+SUPPORTED_STREAM_SOURCES = {"KafkaSource", "PushSource"}
+
+
+class Aggregation(abc.ABC):
+    """
+    NOTE: Feast-handled aggregations are not yet supported. This class provides a way to register user-defined aggregations.
+    """
+
+    column: str  # Column name of the feature we are aggregating.
```

Comment:
nit: would prefer these comments be in an Attributes section, see e.g. `field.py`